crates/cli/src/subcommands/login.rs

```diff
@@ -120,43 +120,76 @@ async fn spacetimedb_token_cached(
 }
 
 async fn web_login_cached(config: &mut Config, host: &str, clear_cache: bool) -> anyhow::Result<String> {
-    let session_id = config.web_session_id().filter(|_| !clear_cache);
-    if let Some(session_id) = session_id {
+    let session_token = config.web_session_token().filter(|_| !clear_cache);
+    if let Some(session_token) = session_token {
         // Currently, these session IDs do not expire. At some point in the future, we may also need to check this session ID for validity.
-        Ok(session_id.clone())
+        Ok(session_token.clone())
     } else {
-        let session_id = web_login(host).await?;
-        config.set_web_session_id(session_id.clone());
+        let session_token = web_login(host).await?;
+        config.set_web_session_token(session_token.clone());
         config.save();
-        Ok(session_id)
+        Ok(session_token)
     }
 }
 
 #[derive(Deserialize)]
-struct WebLoginTokenResponse {
+struct WebLoginTokenData {
     token: String,
 }
 
+#[derive(Deserialize)]
+struct WebLoginTokenResponse {
+    success: bool,
+    data: WebLoginTokenData,
+}
+
 #[derive(Deserialize)]
 struct WebLoginSessionResponse {
+    success: bool,
+    error: Option<String>,
+    data: Option<WebLoginSessionData>,
+}
+
+#[derive(Deserialize)]
+struct WebLoginSessionData {
     approved: bool,
-    session: Option<String>,
+
+    #[serde(rename = "sessionToken")]
+    session_token: Option<String>,
 }
 
 #[derive(Deserialize)]
 struct WebLoginSessionResponseApproved {
-    session_id: String,
+    session_token: String,
 }
 
 impl WebLoginSessionResponse {
-    fn approved(&self) -> Option<WebLoginSessionResponseApproved> {
-        if self.approved {
-            Some(WebLoginSessionResponseApproved {
-                session_id: self.session.clone().unwrap(),
-            })
-        } else {
-            None
+    fn approved(&self) -> Result<Option<WebLoginSessionResponseApproved>, String> {
+        if !self.success {
+            // If success is false, return the error message if available
+            return Err(self.error.clone().unwrap_or("Unknown error".to_string()));
         }
+
+        // If success is true, check for the approved status and session_token
+        if let Some(data) = &self.data {
+            if data.approved {
+                // Approved is true, create the approved response with session_token
+                if let Some(session_token) = &data.session_token {
+                    return Ok(Some(WebLoginSessionResponseApproved {
+                        session_token: session_token.clone(),
+                    }));
+                } else {
+                    // If approved is true but session_token is missing, return an error
+                    return Err("Session token is missing in response.".to_string());
+                }
+            } else {
+                // Approved is false, no session token expected
+                return Ok(None);
+            }
```

Comment:
nit: I don't think we should be using both `if-else` cascades _and_ early return - can you do one or the other?

Reply:
Makes sense. I'm turning in for the day and won't be able to change it for next 12 hours or so. If you have bandwidth, feel free to give it a go